OpenRA.Mods.Common/Widgets/Logic/AssetBrowserLogic.cs

```diff
@@ -397,41 +452,75 @@ bool LoadAsset(IReadOnlyPackage package, string filename)
 						prefix += "|";
 				}
 
-				var video = VideoLoader.GetVideo(Game.ModData.DefaultFileSystem.Open(filename), Game.ModData.VideoLoaders);
-				if (video != null)
+				var fileExtension = Path.GetExtension(filename.ToLowerInvariant());
+				if (allowedSpriteExtensions.Contains(fileExtension))
 				{
-					player = panel.Get<VideoPlayerWidget>("PLAYER");
-					player.Load(prefix + filename);
-					player.DrawOverlay = false;
-					isVideoLoaded = true;
+					currentSprites = world.Map.Rules.Sequences.SpriteCache[prefix + filename];
+					currentFrame = 0;
 
 					if (frameSlider != null)
 					{
-						frameSlider.MaximumValue = (float)player.Video.Frames - 1;
-						frameSlider.Ticks = 0;
+						frameSlider.MaximumValue = (float)currentSprites.Length - 1;
+						frameSlider.Ticks = currentSprites.Length;
 					}
 
-					return true;
-				}
+					currentVoxel = null;
 
-				if (Path.GetExtension(filename.ToLowerInvariant()) == ".vxl")
+					// Just in case we're switching away from a type of asset that forced the music to pause.
+					UnMuteSounds();
+				}
+				else if (allowedModelExtensions.Contains(fileExtension))
 				{
 					var voxelName = Path.GetFileNameWithoutExtension(filename);
 					currentVoxel = world.ModelCache.GetModel(voxelName);
 					currentSprites = null;
+
+					// Just in case we're switching away from a type of asset that forced the music to pause.
+					UnMuteSounds();
 				}
-				else
+				else if (allowedAudioExtensions.Contains(fileExtension))
 				{
-					currentSprites = world.Map.Rules.Sequences.SpriteCache[prefix + filename];
-					currentFrame = 0;
-
-					if (frameSlider != null)
+					// Mute music so it doesn't interfere with current asset.
+					MuteSounds();
+
+					// For whatever reason disposing of this stream immediately still lets sound play in all mods except D2k.
+					var soundStream = Game.ModData.DefaultFileSystem.Open(prefix + filename);
+
+					foreach (var modDataSoundLoader in Game.ModData.SoundLoaders)
+						if (modDataSoundLoader.TryParseSound(soundStream, out currentSoundFormat))
```

Comment:
Skipping the braces here is a bit too much imho.

Reply:
We require braces for the outer `for` if the inner block uses braces.